src/deploy.ts

```diff
@@ -445,6 +456,31 @@ export async function deploy(
   Telemetry.record({event: "deploy", step: "finish"});
 }
 
+async function findOldestSourceAge(effects: DeployEffects, config: Config): Promise<number> {
+  let oldestAge = -Infinity;
+  const nowMs = Date.now();
```

Comment:
I would prefer if this function returned the oldest modification time rather than the “age” relative to now. And same for `findBuildFiles`. That would let us use the same _now_ in both cases, but also I think it’s easier to reason about timestamps than relative times. That would slightly increase the scope of this PR though.